### PR TITLE
Issue1_RedeclareReals

### LINKED ISSUE #1: style: Declaration of Variables
Following rule 3.8 we should use units for variables, that have a physical correspondence - and if possible use types from Modelica.SIUnits. 

There exists lot of variables in the BuildingSystems, that are declared as Real (without a unit) but have a physical correspondece. 
___
Following files contain the keyword `Real` and were NOT copied from Annex60. Files from package `Types` some subpackages of `Utilities` are not considered as they do not use physical quantities:
___
- [x] buildings\airvolumes\airvolumecompressible0d.mo
- [x] buildings\airvolumes\airvolume3d.mo
- [x] buildings\baseclasses\airpathgeneral.mo
- [x] buildings\baseclasses\relationradiationconvection.mo
- [x] buildings\baseclasses\zonetemplategeneral.mo
- [x] buildings\buildingtemplates\building1zone0d_vis3d.mo
- [x] buildings\buildingtemplates\building1zone0ddistrict_vis3d.mo
- [x] buildings\buildingtemplates\building1zone1dcylinder_vis3d.mo
- [x] buildings\buildingtemplates\building1zone0ddistrict.mo
- [x] buildings\constructions\walls\wallthermal3d_vis3d.mo
- [x] buildings\constructions\walls\wallthermal1dnodes_vis3d.mo
- [x] buildings\constructions\windows\radiationtransmission.mo
- [x] buildings\constructions\windows\window.mo
- [x] buildings\constructions\windows\window_vis3d.mo
- [x] buildings\data\airpathcharacteristics\airpathcharacteristicwindow.mo
- [x] buildings\data\airpathcharacteristics\cdtiltedpane.mo
- [x] buildings\zones\zonetemplateairvolumemixed.mo
- [x] buildings\airpathes\airpathconstant.mo
- [x] buildings\airpathes\airpathprescribedpressure.mo
- [x] buildings\functions\tair.mo
- [x] buildings\functions\tsky.mo
- [x] buildings\functions\pair.mo
- [x] buildings\functions\vwind.mo
- [x] buildings\interfaces\airpathportgeneral.mo
- [x] buildings\ambient.mo
- [x] climate\solarradiationtransformers\solarradiationtransformergeneral.mo
- [x] fluid\fixedresistances\pipe.mo
- [x] ham\convectiveheattransfer\surfaces\alphaplatefree.mo
- [x] ham\data\materialproperties\baseclasses\materialhygrothermalgeneral.mo
- [x] ham\heatandmoisturetransport\baseclasses\heatandmoisturetransportgeneral.mo
- [x] ham\heatandmoisturetransport\functions\examples\dphi.mo
- [x] ham\heatandmoisturetransport\functions\dphi.mo
- [x] ham\heatandmoisturetransport\functions\interpol.mo
- [x] ham\heatandmoisturetransport\functions\phisor.mo
- [x] ham\heatandmoisturetransport\functions\wderphi.mo
- [x] ham\heatandmoisturetransport\functions\wsor.mo
- [x] ham\heatandmoisturetransport\heatandmoisturetransport1d.mo
- [x] ham\heatandmoisturetransport\heatandmoisturetransport1dnodes.mo
- [x] ham\heatandmoisturetransport\heatandmoisturetransport3d.mo
- [x] ham\heatconduction\heatconduction1d_vis3d.mo
- [x] ham\heatconduction\heatconduction3d.mo
- [x] ham\heatconduction\heatconduction1d.mo
- [x] technologies\chillers\data\compressionchillers\compressionchillergeneral.mo
- [x] technologies\chillers\compressionchiller.mo
- [x] technologies\electricalstorages\batterysimple.mo
- [x] technologies\photovoltaics\baseclasses\electricalmodels\electricalmodelonediodegeneral.mo
- [x] technologies\photovoltaics\baseclasses\electricalmodels\electricalmodeltwodiodesgeneral.mo
- [x] technologies\photovoltaics\baseclasses\electricalmodels\electricalmodeltwodiodesmpp.mo
- [x] technologies\photovoltaics\baseclasses\electricalmodels\electricalmodelgeneral.mo
- [x] technologies\photovoltaics\baseclasses\thermalmodels\thermalmodelsimple.mo
- [x] technologies\photovoltaics\baseclasses\pvmodulegeneral.mo
- [x] technologies\photovoltaics\data\photovoltaicmodules.mo
- [x] technologies\photovoltaics\pvmodules\pvmodulesimple.mo
- [x] technologies\photovoltaics\pvmodules\pvmodulesimplempp.mo
- [x] technologies\solarthermal\data\collectors\collectorpartial.mo
- [x] technologies\solarthermal\thermalcollector.mo
- [x] technologies\thermalstorages\baseclasses\buoyancymodels\buoyancy1.mo
- [x] technologies\thermalstorages\baseclasses\buoyancymodels\buoyancy2.mo
- [x] technologies\thermalstorages\baseclasses\stratificationmodel\partialinletstratification.mo
- [x] utilities\moistairfunctions\t_sat.mo
- [x] utilities\moistairfunctions\enthalpy.mo
- [x] utilities\moistairfunctions\p.mo
- [x] utilities\moistairfunctions\p_sat.mo
- [x] utilities\moistairfunctions\t_dp.mo

### PR DESCRIPTION
**Pull request by jmoeckel**
This should fix #1 

Several variables declared as `Real` have been redaclered with a physical type. 
Some new types in `BuildingSystems.Types` have been defined.

### CODE REVIEW

**Comment by nytschgeusen** [comment]
@jmoeckel: I agree in principal with the content of the pull request, but there are some version conflicts (the pull request cannot be automatically merged). Please can you solve these conflicts and do the pull request again afterwards ?

**Comment by jmoeckel** [comment]
@nytschgeusen: Merge conflicts have been solved. Branch can now be merged automatically.

File: BuildingSystems/Buildings/Airvolumes/Airvolume3D.mo [hunk 1]
```diff
@@ -660,7 +660,7 @@ parameter Modelica.SIunits.Length width
         "Air density under nominal conditions" annotation (HideResult=true);
       constant Modelica.SIunits.Temp_K T_nominal = 293.15
         "Air temperature under nominal conditions";
-      constant Real rH2O(unit="J/kg") = Medium.enthalpyOfVaporization(T_nominal)
+      constant BuildingSystems.Types.WaterVapourEnthalpy rH2O(unit="J/kg") = Medium.enthalpyOfVaporization(T_nominal)
```
Comment (thorade) [inline BuildingSystems/Buildings/Airvolumes/Airvolume3D.mo]: The type already defines the unit, do not try to re-define it here!

File: BuildingSystems/Types/WaterVapourGasConstant.mo [hunk 2]
```diff
@@ -0,0 +1,2 @@
+within BuildingSystems.Types;
+type WaterVapourGasConstant = Real(final quantity="Gas Constant", final unit="J/(kg.K)");
```
Comment (thorade) [inline BuildingSystems/Types/WaterVapourGasConstant.mo]: Don't re-define the unit (even if it is identical).

**Comment by jmoeckel** [comment]
Partly. 

The other reason has been, that `type WaterVapourEnthalpie = SpecificEnergy;` leads to `Class not found` errors.
Corrected this one to `type WaterVapourEnthalpie = Modelica.SIunits.SpecificEnergy;`

**Comment by jmoeckel** [comment]
Should be fixed with #16

**Pull request merged**